src/shared/install.c

```diff
@@ -1720,19 +1720,65 @@ static int install_info_symlink_alias(
         assert(config_path);
 
         STRV_FOREACH(s, i->aliases) {
+
                 _cleanup_free_ char *alias_path = NULL, *dst = NULL;
 
                 q = install_full_printf(i, *s, &dst);
                 if (q < 0)
                         return q;
 
+                if (unit_name_is_valid(i->name, UNIT_NAME_TEMPLATE) && !unit_name_is_valid(dst, UNIT_NAME_TEMPLATE) && !i->default_instance)
```

Comment:
Line breaks?

I think the code would be much easier to read if it was structured as
```C
   if (... UNIT_NAME_TEMPLATE ...)  {
           if (!unit_name_is_valid(dst, UNIT_NAME_TEMPLATE) && !i->default_instance))
                   ...;
   } else if (... UNIT_NAME_INSTANCE ...))  {
          if (...)
                  ...;
  }
```

Reply:
unfortunately transforming the condition into multiple if doesn't work, the else if in you example wouldn't be taken for templates and it needs to be taken .

I have split the condition into multiple lines, for readability